File: ticketProject5-DistributedC/TicketSystemCC/model/model/Validators/ARValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace model.Validators
{
    /**
     * Validates an instance of class ArtisticRepresentation
     * Throws ValidateException if instance is invalid
     */
    public class ARValidator : Validator<ArtisticRepresentation>
    {
        virtual public void Validate(ArtisticRepresentation entity)
        {
            if (entity == null)
                throw new ArgumentNullException("entity must be not null");
            else
            {
                if (entity.Id <= 0)
                    throw new ValidationException("Invalid ID!");
                if (entity.ArtistName == null)
                    throw new ValidationException("Invalid artist name!");
                if (entity.Data == null)
                    throw new ValidationException("Invalid data!");
                if (entity.Location == null)
                    throw new ValidationException("Invalid location!");
                if (entity.AvailableSeats < 0)
                    throw new ValidationException("Invalid number of available seats!");
                if (entity.SoldSeats < 0)
                    throw new ValidationException("Invalid number of sold seats!");
            }
        }
    }
}

File: ticketProject9-REST/TS-REST/TicketModel/src/main/java/ticket/model/validators/ARValidator.java
package ticket.model.validators;

import ticket.model.ArtisticRepresentation;

import java.security.InvalidParameterException;


/**
 * Validates an instance of class ArtisticRepresentation
 * Throws ValidateException if instance is invalid
 */
public class ARValidator implements Validator<ArtisticRepresentation> {

    /**
     * @throws InvalidParameterException if id  id is null or <=0 / first name/last name are null.
     */
    @Override
    public void validate(ArtisticRepresentation entity) throws ValidationException {
        if (entity == null)
            throw new InvalidParameterException();
        else {
            if (entity.getId() <= 0)
                throw new ValidationException("Invalid ID!");
            if (entity.getArtistName() == null)
                throw new ValidationException("Invalid first name!");
            if (entity.getLocation() == null)
                throw new ValidationException("Invalid location!");
            if (entity.getData() == null)
                throw new ValidationException("Invalid data!");
            if (entity.getAvailableSeats() < 0)
                throw new ValidationException("Invalid number of available seats!");
            if (entity.getSoldSeats() < 0)
                throw new ValidationException("Invalid number of sold seats!");
        }
    }
}
